### DISCUSSION TITLE
Query param of type UUID is presented as array[string] instead of array[string($uuid)] when capturing lists

### DISCUSSION THREAD

**Discussion by josecsotomorales** [fastapi/fastapi#6483] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI

app = FastAPI()

@app.get("/test")
def get_query_params(
                  field_1: Optional[UUID] = None, # type on swagger: string($uuid) - OK
                  field_2: Optional[List[str]] = Query(None), # type on swagger: array[string] - OK
                  field_3: Optional[List[UUID]] = Query(None) # type on swagger: array[string] - NOT OK, Pydantic still validates it as an uuid.
                  ):
    return {"Hello", "World"}
```

### Description

Open the browser and navigate to swagger docs on `http://localhost:8000/docs`, then go to the endpoint parameter types, which should be enough to reproduce this issue.

### Operating System

macOS

### FastAPI Version

0.70.1

### Python Version

3.9.9

**Comment by ugtar** [comment, marked as answer]
it appears to be a purely cosmetic issue in swagger ui, not a fastapi bug. I created a similar setup
```python
from typing import Optional, List
import uuid

from fastapi import FastAPI, Query


app = FastAPI()


@app.get("/")
async def root(
    field0: Optional[uuid.UUID] = None,
    field: Optional[List[uuid.UUID]] = Query(default=None)
):
    return {"message": "Hello World"}
```
The corresponding swagger ui looks like this
![image](https://user-images.githubusercontent.com/96333/147783904-35444999-f2f3-4e6b-9ee4-1e1e918d7589.png)
as you said the description in swagger ui for the list is `array[string]` not `array[string($uuid)]`, but the field is validated properly as a uuid in swagger ui
![image](https://user-images.githubusercontent.com/96333/147783982-1f11d417-8883-4921-a35d-5a91b5b7ff37.png)
as well as correctly validated by pydantic
```
curl -X 'GET'   'http://127.0.0.1:8000/?field=3fa85f64-5717-4562-b3fc-2c963f66afa&field=1d27e9ac-238c-4a8b-b988-b7f5d2471318'   -H 'accept: application/json'
{"detail":[{"loc":["query","field",0],"msg":"value is not a valid uuid","type":"type_error.uuid"}]}
```
The openapi also appears correct
```json
  "paths": {
    "/": {
      "get": {
        "summary": "Root",
        "operationId": "root__get",
        "parameters": [
          {
            "required": false,
            "schema": {
              "title": "Field0",
              "type": "string",
              "format": "uuid"
            },
            "name": "field0",
            "in": "query"
          },
          {
            "required": false,
            "schema": {
              "title": "Field",
              "type": "array",
              "items": {
                "type": "string",
                "format": "uuid"
              }
            },
            "name": "field",
            "in": "query"
          }
        ],
```
You can see each item is `type: string` and `format: uuid`.

So this looks like a cosmetic issue in swagger ui, not a behavior issue in either swagger or fastapi.

**Comment by josecsotomorales** [comment]
@ugtar thanks for your response! you are correct! it all works as expected, validation, formatting etc, seems to be purely visual from swagger docs, closing this issue.

**Comment by ugtar** [comment]
...but you might still open an issue on https://github.com/swagger-api/swagger-ui
